File: Src/Visual Basic .NET/Multiple/Xceed Windows Explorer/TemporaryFileCopyActionForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Xceed.FileSystem.Samples
  Public Class TemporaryFileCopyActionForm : Inherits System.Windows.Forms.Form
    #Region "CONSTRUCTORS"

    Public Sub New()
      InitializeComponent()
    End Sub

    #End Region ' CONSTRUCTORS

    #Region "PUBLIC METHODS"

    Public Overloads Function ShowDialog(ByVal owner As IWin32Window, <System.Runtime.InteropServices.Out()> ByRef fileOnly As Boolean, <System.Runtime.InteropServices.Out()> ByRef recursive As Boolean) As DialogResult
      fileOnly = True
      recursive = False

      Dim result As DialogResult = Me.ShowDialog(owner)

      If result = System.Windows.Forms.DialogResult.OK Then
        If FileOnlyOption.Checked Then
          fileOnly = True
          recursive = False
        Else
          fileOnly = False
          recursive = FileAndFolderRecursiveOption.Checked
        End If
      End If

      Return result
    End Function

    #End Region ' PUBLIC METHODS

    #Region "PROTECTED METHODS"

    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
      If disposing Then
        If Not components Is Nothing Then
          components.Dispose()
        End If
      End If
      MyBase.Dispose(disposing)
    End Sub

    #End Region ' PROTECTED METHODS

    #Region "Windows Form Designer generated code"
    ''' <summary>
    ''' Required method for Designer support - do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>
    Private Sub InitializeComponent()
      Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(TemporaryFileCopyActionForm))
      Me.label1 = New System.Windows.Forms.Label()
      Me.FileOnlyOption = New System.Windows.Forms.RadioButton()
      Me.FileAndFolderNonRecursiveOption = New System.Windows.Forms.RadioButton()
      Me.FileAndFolderRecursiveOption = New System.Windows.Forms.RadioButton()
      Me.OkBtn = New System.Windows.Forms.Button()
      Me.CancelBtn = New System.Windows.Forms.Button()
      Me.SuspendLayout()
      ' 
      ' label1
      ' 
      Me.label1.Location = New System.Drawing.Point(16, 16)
      Me.label1.Name = "label1"
      Me.label1.Size = New System.Drawing.Size(312, 40)
      Me.label1.TabIndex = 0
      Me.label1.Text = "In order to complete this operation, some file(s) needs to be copied locally. Sel" & "ect which files you want to copy to a temporary folder."
      ' 
      ' FileOnlyOption
      ' 
      Me.FileOnlyOption.Checked = True
      Me.FileOnlyOption.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.FileOnlyOption.Location = New System.Drawing.Point(32, 64)
      Me.FileOnlyOption.Name = "FileOnlyOption"
      Me.FileOnlyOption.Size = New System.Drawing.Size(288, 24)
      Me.FileOnlyOption.TabIndex = 0
      Me.FileOnlyOption.TabStop = True
      Me.FileOnlyOption.Text = "Selected file only"
      ' 
      ' FileAndFolderNonRecursiveOption
      ' 
      Me.FileAndFolderNonRecursiveOption.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.FileAndFolderNonRecursiveOption.Location = New System.Drawing.Point(32, 88)
      Me.FileAndFolderNonRecursiveOption.Name = "FileAndFolderNonRecursiveOption"
      Me.FileAndFolderNonRecursiveOption.Size = New System.Drawing.Size(288, 24)
      Me.FileAndFolderNonRecursiveOption.TabIndex = 1
      Me.FileAndFolderNonRecursiveOption.Text = "Selected file and folder content (non-recursive)"
      ' 
      ' FileAndFolderRecursiveOption
      ' 
      Me.FileAndFolderRecursiveOption.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.FileAndFolderRecursiveOption.Location = New System.Drawing.Point(32, 112)
      Me.FileAndFolderRecursiveOption.Name = "FileAndFolderRecursiveOption"
      Me.FileAndFolderRecursiveOption.Size = New System.Drawing.Size(288, 24)
      Me.FileAndFolderRecursiveOption.TabIndex = 2
      Me.FileAndFolderRecursiveOption.Text = "Selected file and folder content (recursive)"
      ' 
      ' OkBtn
      ' 
      Me.OkBtn.DialogResult = System.Windows.Forms.DialogResult.OK
      Me.OkBtn.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.OkBtn.Location = New System.Drawing.Point(160, 152)
      Me.OkBtn.Name = "OkBtn"
      Me.OkBtn.Size = New System.Drawing.Size(80, 23)
      Me.OkBtn.TabIndex = 3
      Me.OkBtn.Text = "&Ok"
      ' 
      ' CancelBtn
      ' 
      Me.CancelBtn.DialogResult = System.Windows.Forms.DialogResult.Cancel
      Me.CancelBtn.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.CancelBtn.Location = New System.Drawing.Point(248, 152)
      Me.CancelBtn.Name = "CancelBtn"
      Me.CancelBtn.Size = New System.Drawing.Size(80, 23)
      Me.CancelBtn.TabIndex = 4
      Me.CancelBtn.Text = "&Cancel"
      ' 
      ' TemporaryFileCopyActionForm
      ' 
      Me.AcceptButton = Me.OkBtn
      Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
      Me.CancelButton = Me.CancelBtn
      Me.ClientSize = New System.Drawing.Size(338, 184)
      Me.Controls.Add(Me.CancelBtn)
      Me.Controls.Add(Me.OkBtn)
      Me.Controls.Add(Me.FileAndFolderRecursiveOption)
      Me.Controls.Add(Me.FileAndFolderNonRecursiveOption)
      Me.Controls.Add(Me.FileOnlyOption)
      Me.Controls.Add(Me.label1)
      Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
      Me.Icon = (CType(resources.GetObject("$this.Icon"), System.Drawing.Icon))
      Me.MaximizeBox = False
      Me.Name = "TemporaryFileCopyActionForm"
      Me.Text = "Copying temporary files"
      Me.ResumeLayout(False)

    End Sub
    #End Region

    #Region "Windows Form Designer generated fields"

    Private label1 As System.Windows.Forms.Label
    Private OkBtn As System.Windows.Forms.Button
    Private FileOnlyOption As System.Windows.Forms.RadioButton
    Private FileAndFolderNonRecursiveOption As System.Windows.Forms.RadioButton
    Private FileAndFolderRecursiveOption As System.Windows.Forms.RadioButton
    Private CancelBtn As System.Windows.Forms.Button
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private components As System.ComponentModel.Container = Nothing

    #End Region ' Windows Form Designer generated fields
  End Class
End Namespace

File: Src/DetailedSamples/CSharp/Multiple/Xceed Windows Explorer/TemporaryFileCopyActionForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace Xceed.FileSystem.Samples
{
	public class TemporaryFileCopyActionForm : System.Windows.Forms.Form
	{
    #region CONSTRUCTORS

		public TemporaryFileCopyActionForm()
		{
			InitializeComponent();
		}

    #endregion CONSTRUCTORS

    #region PUBLIC METHODS

    public DialogResult ShowDialog( IWin32Window owner, out bool fileOnly, out bool recursive )
    {
      fileOnly = true;
      recursive = false;

      DialogResult result = this.ShowDialog( owner );

      if( result == DialogResult.OK )
      {
        if( FileOnlyOption.Checked )
        {
          fileOnly = true;
          recursive = false;
        }
        else 
        {
          fileOnly = false;
          recursive = FileAndFolderRecursiveOption.Checked;
        }
      }

      return result;
    }

    #endregion PUBLIC METHODS

    #region PROTECTED METHODS

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

    #endregion PROTECTED METHODS

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
      System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(TemporaryFileCopyActionForm));
      this.label1 = new System.Windows.Forms.Label();
      this.FileOnlyOption = new System.Windows.Forms.RadioButton();
      this.FileAndFolderNonRecursiveOption = new System.Windows.Forms.RadioButton();
      this.FileAndFolderRecursiveOption = new System.Windows.Forms.RadioButton();
      this.OkBtn = new System.Windows.Forms.Button();
      this.CancelBtn = new System.Windows.Forms.Button();
      this.SuspendLayout();
      // 
      // label1
      // 
      this.label1.Location = new System.Drawing.Point(16, 16);
      this.label1.Name = "label1";
      this.label1.Size = new System.Drawing.Size(312, 40);
      this.label1.TabIndex = 0;
      this.label1.Text = "In order to complete this operation, some file(s) needs to be copied locally. Sel" +
        "ect which files you want to copy to a temporary folder.";
      // 
      // FileOnlyOption
      // 
      this.FileOnlyOption.Checked = true;
      this.FileOnlyOption.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.FileOnlyOption.Location = new System.Drawing.Point(32, 64);
      this.FileOnlyOption.Name = "FileOnlyOption";
      this.FileOnlyOption.Size = new System.Drawing.Size(288, 24);
      this.FileOnlyOption.TabIndex = 0;
      this.FileOnlyOption.TabStop = true;
      this.FileOnlyOption.Text = "Selected file only";
      // 
      // FileAndFolderNonRecursiveOption
      // 
      this.FileAndFolderNonRecursiveOption.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.FileAndFolderNonRecursiveOption.Location = new System.Drawing.Point(32, 88);
      this.FileAndFolderNonRecursiveOption.Name = "FileAndFolderNonRecursiveOption";
      this.FileAndFolderNonRecursiveOption.Size = new System.Drawing.Size(288, 24);
      this.FileAndFolderNonRecursiveOption.TabIndex = 1;
      this.FileAndFolderNonRecursiveOption.Text = "Selected file and folder content (non-recursive)";
      // 
      // FileAndFolderRecursiveOption
      // 
      this.FileAndFolderRecursiveOption.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.FileAndFolderRecursiveOption.Location = new System.Drawing.Point(32, 112);
      this.FileAndFolderRecursiveOption.Name = "FileAndFolderRecursiveOption";
      this.FileAndFolderRecursiveOption.Size = new System.Drawing.Size(288, 24);
      this.FileAndFolderRecursiveOption.TabIndex = 2;
      this.FileAndFolderRecursiveOption.Text = "Selected file and folder content (recursive)";
      // 
      // OkBtn
      // 
      this.OkBtn.DialogResult = System.Windows.Forms.DialogResult.OK;
      this.OkBtn.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.OkBtn.Location = new System.Drawing.Point(160, 152);
      this.OkBtn.Name = "OkBtn";
      this.OkBtn.Size = new System.Drawing.Size(80, 23);
      this.OkBtn.TabIndex = 3;
      this.OkBtn.Text = "&Ok";
      // 
      // CancelBtn
      // 
      this.CancelBtn.DialogResult = System.Windows.Forms.DialogResult.Cancel;
      this.CancelBtn.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.CancelBtn.Location = new System.Drawing.Point(248, 152);
      this.CancelBtn.Name = "CancelBtn";
      this.CancelBtn.Size = new System.Drawing.Size(80, 23);
      this.CancelBtn.TabIndex = 4;
      this.CancelBtn.Text = "&Cancel";
      // 
      // TemporaryFileCopyActionForm
      // 
      this.AcceptButton = this.OkBtn;
      this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
      this.CancelButton = this.CancelBtn;
      this.ClientSize = new System.Drawing.Size(338, 184);
      this.Controls.Add(this.CancelBtn);
      this.Controls.Add(this.OkBtn);
      this.Controls.Add(this.FileAndFolderRecursiveOption);
      this.Controls.Add(this.FileAndFolderNonRecursiveOption);
      this.Controls.Add(this.FileOnlyOption);
      this.Controls.Add(this.label1);
      this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
      this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
      this.MaximizeBox = false;
      this.Name = "TemporaryFileCopyActionForm";
      this.Text = "Copying temporary files";
      this.ResumeLayout(false);

    }
		#endregion

    #region Windows Form Designer generated fields

    private System.Windows.Forms.Label label1;
    private System.Windows.Forms.Button OkBtn;
    private System.Windows.Forms.RadioButton FileOnlyOption;
    private System.Windows.Forms.RadioButton FileAndFolderNonRecursiveOption;
    private System.Windows.Forms.RadioButton FileAndFolderRecursiveOption;
    private System.Windows.Forms.Button CancelBtn;
    /// <summary>
    /// Required designer variable.
    /// </summary>
    private System.ComponentModel.Container components = null;

    #endregion Windows Form Designer generated fields
	}
}
